Styling and Position per theme

azzaromax opened the pull request:
switcher for modal view in admin
Added SW (but not work)

azzaromax committed:
Styling and Position per theme
switcher for modal view in admin
Added SW (but not work)

azzaromax committed:
Load theme config then module enabled
Added rout for load terms
Some changes
Fix problem with save config
Show libraries only on the atm config page.
Remove atm.min.js before save new file.
Added SW
Added checking for generate styles
Get config without cache
On change theme retrieve theme or create new
Added selecting CT

AlexanderC commented on atm.module:
```diff
@@ -83,16 +82,87 @@ function atm_theme($existing, $type, $theme, $path) {
+    if ($routName == 'atm.config.page' || in_array($node->getType(), $helper->getSelectedContentTypes())) {
+      $atmApi = $helper->get('build_path') . '?' . microtime();
+
+      $https = &$_SERVER['HTTPS'];
```
Please set https check to `!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' || $_SERVER['SERVER_PORT'] == 443` this may fix issues on ISAPI with IIS

azzaromax replied:
Done
https://github.com/AdTechMedia/drupal-module/pull/50/files#diff-07b0b3a19f4033b9a69fd7b018eed316R101

AlexanderC commented on config/install/atm.settings.yml:
```diff
@@ -1,6 +1,8 @@
 register_url: https://www.adtechmedia.io/admin/accounts/signup
-#base_endpoint: https://api-dev.adtechmedia.io/v1
-base_endpoint: https://api.adtechmedia.io/v1/
+base_endpoint: https://api-dev.adtechmedia.io/v1
```
Please use prod env by default: `https://api.adtechmedia.io/v1`

azzaromax replied:
Done
https://github.com/AdTechMedia/drupal-module/pull/50/files#diff-5f4cf980dd1446415e2e55add63f3348R2

AlexanderC commented on js/admin.js:
```diff
@@ -36,18 +79,25 @@
+        var output = atmTemplating.templateRendition(componentName + 'Component').render(options, styles);
+      } catch (Error) {
+
+        console.log(Error);
```
Please manage rendition exception (e.g. missing component) and improve UX

azzaromax replied:
Done
https://github.com/AdTechMedia/drupal-module/pull/50/files#diff-a0623ef93657e56554c5ddda522ed690R100

AlexanderC commented on src/Controller/AtmApiController.php:
```diff
@@ -16,10 +19,67 @@ class AtmApiController extends ControllerBase {
+    $https = &$_SERVER['HTTPS'];
+    if ($https == 'on') {
+      return new Response(
+        file_get_contents('https://api.adtechmedia.io/atm-admin/atm-build/sw.min.js'),
```
Could we move this to configuration? `https://api.adtechmedia.io/atm-admin/atm-build/sw.min.js`

AlexanderC commented on src/Controller/AtmApiController.php:
```diff
@@ -16,10 +19,67 @@ class AtmApiController extends ControllerBase {
+    }
+    else {
+      try {
+        $response = \Drupal::httpClient()->get('https://www.adtechmedia.io/terms/dialog.html');
```
Could we move this to configuration? `https://www.adtechmedia.io/terms/dialog.html`

azzaromax committed:
Fixed bug with scroll top after ajax
Moved selecting CT to content config form
Update atm.js on frontend on atm config page after saving styles

azzaromax committed:
Mover terms url to settings
Changes endpoint to prod version
Show notify if error on edit inputs with vars
Added cron for regenearte property config

azzaromax committed:
Check https for loading atm.js

azzaromax committed:
Change text in template state on admin config page

azzaromax committed:
Show cute notification after ajax

azzaromax committed:
New params for default settings

azzaromax committed:
Added ajax command noty

azzaromax committed:
Addeed bin folder

azzaromax committed:
Added styled switch instead standard checkbox for sticky

azzaromax committed:
remove bin/test

azzaromax committed:
Resolved problem with cache
Fixed main border

azzaromax committed:
bin tests

azzaromax committed:
Issue 49: Fixed. Added position relative for root if no sticky

azzaromax committed:
rm bin folder

azzaromax committed:
return bin folder

eistrati committed:
fix permissions

azzaromax committed:
React applying styles from `OVERALL POSITION AND STYLING` form

azzaromax committed:
Fixed align styles for preview on the config page

azzaromax committed:
Added label for sticky form element

azzaromax committed:
Set `article` content type by default

azzaromax committed:
Show atm modal between title and text

azzaromax committed:
Merge branch 'new-module-version' of https://github.com/AdTechMedia/drupal-module into new-module-version
